spi/src/main/java/org/opensearch/jobscheduler/spi/LockModel.java

```diff
@@ -55,6 +63,45 @@
     private final boolean released;
     private final long seqNo;
     private final long primaryTerm;
+    private final boolean isResourceLock;
```

Comment:
I see we are not storing this - is this a proxy field tracking `resource == null` for anyone interacting with lockModel instance?